File: hackerrank/30-days-of-code/binary_numbers/binary_numbers.cpp
#include <bits/stdc++.h>

using namespace std;

// string convertToBinary(int n);

std::string convertToBinary(int n) {
    std::string binary;
    int remainder;

    while (n > 0){
        remainder = n % 2;
        n = (n / 2);
        std::string s = to_string(remainder);
        // cout << remainder << endl;
        binary += s;
    }
    return binary;
}

int main()
{
    int n;
    cin >> n;
    cin.ignore(numeric_limits<streamsize>::max(), '\n');
    std::string s = convertToBinary(n);
    // cout << s << endl;
    // std::string s = std::to_string(binaryNumber);

    int counter = 0;
    int j = 0;

    int length = s.size();

    for (int i = 0; i < length; i++)
    {
        if (s[i] == '1'){
            j++;
        }
        else {
            j = 0;
        }
        if (j > counter) {
            counter = j;
        }
    }
    cout << counter << endl;

    return 0;
}

File: hackerrank/30-days-of-code/binary_numbers/binary_numbers.py
#!/usr/bin/env python

import math
import os
import random
import re
import sys

def binary_number(num):
    return bin(n).replace('0b', "")

if __name__ == '__main__':
    n = int(input())
    test = []
    counter = 0
    binary = binary_number(n)
    binary = str(binary) + '0'
    for i in binary:
        if i == '1':
            counter += 1
        elif i == '0':
            test.append(counter)
            counter = 0
    # print(''.join(binary))
    # print(test)
    print(max(test))
    